### PR TITLE
[AKS] `az aks create/update`: Update the VM SKU details from backend for validations during Azure Container Storage operations

### PR DESCRIPTION
**Pull request by mukhoakash**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Review by FumingZhang — COMMENTED** [review]
LGTM

But I think the title of PR should be changed, as it's not perceptible to users.

**Review by FumingZhang — COMMENTED** [review]
Please fix failed CI checks (both test cases and style check).

Is there any existing test case could help validate the change? If so, please queue [live test pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=106356084&view=results) to help validate, otherwise, please add a new one.

File: src/azure-cli/setup.py [hunk 1]
```diff
@@ -129,6 +129,7 @@
     'azure-synapse-artifacts~=0.19.0',
     'azure-synapse-managedprivateendpoints~=0.4.0',
     'azure-synapse-spark~=0.2.0',
+    'cachetools~=5.5.0',
```
Comment (FumingZhang) [inline src/azure-cli/setup.py:132]: Would like to get some feedback from the cli team, is it appropriate to add a dependency like this? @zhoxing-ms, @yanzhudd

**Comment by mukhoakash** [comment]
Queued [live tests](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=106452916&view=results).

**Comment by mukhoakash** [comment]
> Please fix failed CI checks (both test cases and style check).
> 
> Is there any existing test case could help validate the change? If so, please queue [live test pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=106356084&view=results) to help validate, otherwise, please add a new one.

Hi @FumingZhang the live tests have run successfully and the failures in the CI are due to unrelated modules.

**Review by FumingZhang — APPROVED** [review]
LGTM

**Comment by yonzhan** [comment]
Please fix CI issues

File: src/azure-cli/setup.py (hunk 1 above)
Comment (bebound) [reply src/azure-cli/setup.py:133]: It seems that a global `{sku_name: (cpu_value, nvme_enabled)}` dict can achieve the same thing? Creating a dummy cached function is too heavy.

Comment (mukhoakash) [reply src/azure-cli/setup.py:133]: Agreed. Modified the implementation to use a global dict.

**Comment by yanzhudd** [comment]
It looks good to me.
@FumingZhang could you please review it again?

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**